gobblin-metrics/src/main/java/gobblin/metrics/KafkaAvroReporter.java

```diff
@@ -0,0 +1,163 @@
+/* (c) 2014 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.metrics;
+
+import com.codahale.metrics.MetricRegistry;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericDatumWriter;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.io.Encoder;
+import org.apache.avro.io.EncoderFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Kafka reporter for codahale metrics writing metrics in Avro format
+ *
+ * @author ibuenros
+ */
+public class KafkaAvroReporter extends KafkaReporter {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(KafkaAvroReporter.class);
+
+  /**
+   * Avro schema used for encoding metrics
+   * TODO: finalize metrics avro schema
+   */
+  private static final String SchemaString = "{\n"+
+      " \"type\": \"record\",\n"+
+      " \"name\": \"Metric\",\n"+
+      " \"namespace\":\"<TBD>\",\n"+
+      " \"fields\" : [\n"+
+      " {\"name\": \"tags\", \"type\": {\"type\": \"map\", \"values\": \"string\"}, \"doc\": \"tags associated with the metric\"},\n"+
+      " {\"name\": \"name\", \"type\": \"string\", \"doc\": \"metric name\"},\n"+
+      " {\"name\": \"value\", \"type\": [\"boolean\", \"int\", \"long\", \"float\", \"double\", \"bytes\", \"string\"], \"doc\": \"metric value\"}\n"+
+      " ]\n"+
+      "}";
+
+  public static final Schema Schema = (new Schema.Parser()).parse(SchemaString);
+  private GenericDatumWriter<GenericRecord> writer;
+  private Encoder encoder;
+  private ByteArrayOutputStream out;
+  private long lastSerializeExceptionTime;
+
+  protected KafkaAvroReporter(Builder<?> builder) {
+    super(builder);
+
+    lastSerializeExceptionTime = 0;
+
+    out = new ByteArrayOutputStream();
```

Comment:
Can you use `this.` whenever you are referring to a member variable, just like elsewhere in the code base.